worlds/x2wotc/mods/lwotc/__init__.py

```diff
@@ -0,0 +1,368 @@
+
+
+name = "Long War of the Chosen"
+
+# For defining the order rules are applied in (in case of set_rule)
+# The order is lowest to highest priority
+rule_priority = 0.0
+
+# Handle mod options here
+def generate_early(world: "X2WOTCWorld"):
+
+    # Weapons have 5 tiers
+    world.item_manager.disable_progressive_item("ProgressiveRifleTechCompleted")
+    world.item_manager.disable_progressive_item("ProgressiveRifleTechCompleted+")
+    if "RifleTech+" in world.options.progressive_items:
+        if not world.item_manager.enable_progressive_item("ProgressiveRifleTechLwotcCompleted+"):
+            print(f"X2WOTC: Failed to enable progressive LWOTC rifle tech+ for player {world.player_name}")
+        if not world.item_manager.enable_progressive_item("ProgressiveAdvancedWeaponTechLwotcCompleted"):
+            print(f"X2WOTC: Failed to enable progressive LWOTC advanced weapon tech for player {world.player_name}")
+    elif "RifleTech" in world.options.progressive_items:
+        if not world.item_manager.enable_progressive_item("ProgressiveRifleTechLwotcCompleted"):
+            print(f"X2WOTC: Failed to enable progressive LWOTC rifle tech for player {world.player_name}")
+        if not world.item_manager.enable_progressive_item("ProgressiveAdvancedWeaponTechLwotcCompleted"):
+            print(f"X2WOTC: Failed to enable progressive LWOTC advanced weapon tech for player {world.player_name}")
+
+    # GREMLINs are upgraded from ADVENT Robotics
+    world.item_manager.disable_progressive_item("ProgressiveGREMLINTechCompleted")
+    if "GREMLINTech" in world.options.progressive_items:
+        if not world.item_manager.enable_progressive_item("ProgressiveGREMLINTechLwotcCompleted"):
+            print(f"X2WOTC: Failed to enable progressive LWOTC GREMLIN tech for player {world.player_name}")
+
+    # Light and Heavy Armor are researches, not PG projects
+    if "ArmorTech" in world.options.progressive_items:
+        if not world.item_manager.enable_progressive_item("ProgressiveLightArmorTechLwotcCompleted"):
+            print(f"X2WOTC: Failed to enable progressive LWOTC light armor tech for player {world.player_name}")
+        if not world.item_manager.enable_progressive_item("ProgressiveHeavyArmorTechLwotcCompleted"):
+            print(f"X2WOTC: Failed to enable progressive LWOTC heavy armor tech for player {world.player_name}")
+
+    # Handle option to force early proving ground
+    if world.options.early_proving_ground:
+        del world.multiworld.early_items[world.player][
+            world.item_manager.item_table["AutopsyAdventOfficerCompleted"].display_name
+        ]
+        world.multiworld.early_items[world.player][
+            world.item_manager.item_table["AutopsyAdventTrooperCompleted"].display_name
+        ] = 1
+
+    # Rocket Launcher is a squaddie Technical skill
+    world.loc_manager.disable_location("UseRocketLauncher")
+
+    # Lost corpses are unobtainable
+    world.loc_manager.disable_location("AutopsyTheLost")
```

Comment:
If you disable `AutopsyTheLost`, you need to disable its `normal_item` too, or you run the risk of ending up with fewer locations than items. As is, this causes a lot of generation failures (that are incorrectly tagged as `OptionError`).
```suggestion
    world.loc_manager.disable_location("AutopsyTheLost")
    world.item_manager.disable_item("AutopsyTheLostCompleted")
```